src/sidebar/util/groups.js

```diff
@@ -20,7 +23,48 @@ function combineGroups(userGroups, featuredGroups) {
   featuredGroups.forEach(group => (group.isMember = false));
   userGroups.forEach(group => (group.isMember = true));
 
-  return userGroups.concat(featuredGroups);
+  const groups = userGroups.concat(featuredGroups);
+
+  // Add isSelectable property indicating whether a group can be
+  // annotated in at this particular url.
+  groups.forEach(group => (group.isSelectable = isSelectable(group, uri)));
+
+  // Remove unneeded scopes data from group to save memory.
+  groups.forEach(group => delete group.scopes);
+
+  return groups;
+}
+
+function isSelectable(group, uri) {
```

Comment:
Might we reflect on this function name? I feel like what this function is determining is whether a given group can have annotations written to it at this document_uri. Being "selectable" is a UI-level concern.